src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -12205,35 +12232,44 @@ LambdaExpressionSyntax parseLambdaExpressionWorker()
                 else
                 {
                     var name = this.ParseIdentifierToken();
-                    SyntaxToken arrow, exclamation;
-                    // Case x! =>
+                    SyntaxToken arrow, exclamationExclamation;
+                    // Case x!! =>
                     if (this.CurrentToken.Kind == SyntaxKind.ExclamationToken)
                     {
-                        exclamation = this.EatToken(SyntaxKind.ExclamationToken);
-                        arrow = this.EatToken(SyntaxKind.EqualsGreaterThanToken);
+                        exclamationExclamation = this.EatToken();
+                        if (this.CurrentToken.Kind == SyntaxKind.ExclamationToken)
+                        {
+                            exclamationExclamation = MergeTokens(exclamationExclamation, this.EatToken(SyntaxKind.ExclamationToken), SyntaxKind.ExclamationExclamationToken);
+                            arrow = this.EatToken(SyntaxKind.EqualsGreaterThanToken);
+                        }
+                        else if (this.CurrentToken.Kind == SyntaxKind.ExclamationEqualsToken)
```

Comment:
Can this actually happen? I wouldn't expect it to, can you point me at a test that shows this?

Reply:
If we're talking about `!!=>` syntax, `NullCheckedLambdaSingleParameterNoSpaces` contains that syntax.

Reply:
It contains the syntax, but does it actually hit this case?

---
In reply to: [467120850](https://github.com/dotnet/roslyn/pull/46520#discussion_r467120850) [](ancestors = 467120850)

Reply:
It does in fact hit this case.